src/linter/ui5Types/bestPractices.ts

```diff
@@ -0,0 +1,358 @@
+	const properties = Object.create(null) as propsRecord;
+
+	node.properties?.forEach((prop) => {
+		if (!ts.isPropertyAssignment(prop) || !prop.name) {
+			return;
+		}
+
+		const key = prop.name.getText();
+		if (prop.initializer.kind === ts.SyntaxKind.TrueKeyword) {
+			properties[key] = {value: true, node: prop.initializer};
+		} else if (prop.initializer.kind === ts.SyntaxKind.FalseKeyword) {
+			properties[key] = {value: false, node: prop.initializer};
+		} else if (prop.initializer.kind === ts.SyntaxKind.NullKeyword) {
+			properties[key] = {value: null, node: prop.initializer};
+		} else if (ts.isObjectLiteralExpression(prop.initializer) && prop.initializer.properties) {
+			properties[key] = {value: extractPropsRecursive(prop.initializer), node: prop.initializer};
+		} else if (ts.isArrayLiteralExpression(prop.initializer)) {
+			const resolvedValue = prop.initializer.elements.map((elem) => {
+				if (!ts.isObjectLiteralExpression(elem)) {
+					return;
+				}
+				return extractPropsRecursive(elem);
+			}).filter(($) => $) as propsRecordValueType[];
+
+			properties[key] = {value: resolvedValue, node: prop.initializer};
+		} else if (
+			(ts.isIdentifier(prop.initializer) ||
+			ts.isNumericLiteral(prop.initializer) ||
+			ts.isStringLiteral(prop.initializer)) &&
+
+			prop.initializer.getText()) {
+			properties[key] = {value: prop.initializer.getText(), node: prop.initializer};
+		} else {
+			// throw new Error("Unhandled property assignment");
+		}
+	});
+	return properties;
+}
+
+function reportResults(
+	analysisResult: AsyncInterfaceFindType,
+	reporter: SourceFileReporter,
+	classDesc: ts.ClassDeclaration
+) {
+	const {hasAsyncInterface, routingAsyncFlag, rootViewAsyncFlag, hasManifestDefinition} = analysisResult;
+
+	if (!hasManifestDefinition) {
+		reporter.addMessage({
+			node: classDesc,
+			severity: LintMessageSeverity.Warning,
+			ruleId: "ui5-linter-add-manifest",
+			message: "Include a manifest section into the Component.js",
+			messageDetails: "manifest.json will be loaded and used, " +
+			"but is not defined in Component's metadata section",
+		});
+	}
+
+	if (hasAsyncInterface !== AsyncInterfaceStatus.true) {
+		if ([AsyncInterfaceStatus.propNotSet, AsyncInterfaceStatus.false].includes(rootViewAsyncFlag) ||
+			[AsyncInterfaceStatus.propNotSet, AsyncInterfaceStatus.false].includes(routingAsyncFlag)) {
+			const componentsToAddress = [];
+			if (AsyncInterfaceStatus.parentPropNotSet !== rootViewAsyncFlag) {
+				componentsToAddress.push("Root View");
+			}
+			if (AsyncInterfaceStatus.parentPropNotSet !== routingAsyncFlag) {
+				componentsToAddress.push("Routing");
+			}
+
+			reporter.addMessage({
+				node: classDesc,
+				severity: LintMessageSeverity.Error,
+				ruleId: "ui5-linter-no-sync-loading",
+				message: `${componentsToAddress.join(" and ")} ${componentsToAddress.length > 1 ? "are" : "is"} ` +
+				"not configured to load targets asynchronously",
```

Comment:
Please review the text. `Root View is not configured to load targets asynchronously` is confusing to me. For the root view there are no "targets", it's just the view. To me it seems this wording only applies to the routing targets.

Reply:
Refactored with https://github.com/SAP/ui5-linter/pull/73/commits/49d4e33e2377f65700613f2e0dbfba68f332d519